### PR TITLE
Pair blacklist

### PR DESCRIPTION
**Pull request by jblestang**
PR to handle #129 and a refactoring of how the pair_whitelist is being handled among the various functions

### CODE REVIEW

File: freqtrade/main.py [hunk 1]
```diff
@@ -26,19 +26,17 @@
 _CONF = {}
 
 
-def refresh_whitelist(whitelist: Optional[List[str]] = None) -> None:
+def refresh_whitelist(whitelist: List[str]) -> List[str]:
     """
     Check wallet health and remove pair from whitelist if necessary
     :param whitelist: a new whitelist (optional)
```
Comment (gcarq) [inline freqtrade/main.py]: :param whitelist description is no longer valid

File: freqtrade/main.py [hunk 2]
```diff
@@ -47,12 +45,10 @@ def refresh_whitelist(whitelist: Optional[List[str]] = None) -> None:
...
                 pair, status.get('Notice') or 'wallet is not active'
             )
-    if _CONF['exchange']['pair_whitelist'] != sanitized_whitelist:
-        logger.debug('Using refreshed pair whitelist: %s ...', sanitized_whitelist)
-        _CONF['exchange']['pair_whitelist'] = sanitized_whitelist
+    return sanitized_whitelist
 
 
-def _process(dynamic_whitelist: Optional[int] = 0) -> bool:
+def _process(nb_assets: Optional[int] = 0) -> bool:
```
Comment (gcarq) [inline freqtrade/main.py]: Docstring needs to be adapted as well

File: freqtrade/tests/test_acl_pair.py [hunk 3]
```diff
@@ -48,12 +48,12 @@ def test_refresh_whitelist(mocker):
     mocker.patch.dict('freqtrade.main._CONF', conf)
     mocker.patch.multiple('freqtrade.main.exchange',
                           get_wallet_health=get_health)
-    # no argument: use the whitelist provided by config
-    refresh_whitelist()
+    refreshedwhitelist = refresh_whitelist(conf['exchange']['pair_whitelist'])
+    print(conf['exchange']['pair_whitelist'])
```
Comment (gcarq) [inline freqtrade/tests/test_acl_pair.py]: These prints shouldn't be necessary.

**Comment by jblestang** [comment]
A network error occured, how can I launch one more time the CI ?

**Comment by gcarq** [comment]
@jblestang LGTM, but please rebase against develop. `assert_list_equal` has been removed and should be replaced with `assert set(list1) == set(list2)`

**Comment by jblestang** [comment]
Done!

**Pull request merged**